### PR TITLE
doc: Updates examples & docs that use replicaSet clusters

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Updates examples & docs that use replicaSet clusters.

- Also simplifies and improves TF linter and validations
- Reduce logs verbosity to less than half in TF validate
- TF linter shows errors in GH directly as comments
- TF validate can now be run in local easily. It uses the [recommended way](https://developer.hashicorp.com/terraform/cli/config/config-file#development-overrides-for-provider-developers) to override the provider

Link to any related issue(s): CLOUDP-260977

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

## Further comments

Example of linter error in GH:

linter_error_gh

### CODE REVIEW

File: .github/tflint-matcher.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+	"problemMatcher": [
```
Comment (lantoli) [inline .github/tflint-matcher.json:2]: to show TF linter errors in GH UI

File: .github/workflows/examples.yml [hunk 2]
```diff
@@ -15,52 +15,27 @@ env:
...
     permissions: {}
     steps:
       - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332
-        with:
-          fetch-depth: 0
-      - run: echo "GO_VERSION=$(cat .go-version)" >> "${GITHUB_ENV}"
       - uses: actions/setup-go@0a12ed9d6a96ab950c8f026ed9f722fe0da7ef32
         with:
           go-version-file: 'go.mod'
-      - name: go build
```
Comment (lantoli) [inline .github/workflows/examples.yml:31]: make it easier to run TF validate in local, so no magic is done in GH Actions and everything happens in the script

File: examples/mongodbatlas_api_key/create-and-assign-pak-together/versions.tf [hunk 3]
```diff
@@ -2,7 +2,7 @@ terraform {
   required_providers {
     mongodbatlas = {
       source  = "mongodb/mongodbatlas"
-      version = "~> 1.13.2"
+      version = "~> 1.0"
```
Comment (lantoli) [inline examples/mongodbatlas_api_key/create-and-assign-pak-together/versions.tf:5]: so it can run with the current provider (version  >= 1, < 2)

File: scripts/tf-validate.tfrc [hunk 4]
```diff
@@ -0,0 +1,6 @@
+provider_installation {
```
Comment (lantoli) [inline scripts/tf-validate.tfrc]: prefer this approximation to override provider with code in master instead of doing multiple string replacements in the examples

**Review by carriecwk — APPROVED** [review]
LGTM

File: docs/data-sources/advanced_clusters.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 # Data Source: mongodbatlas_clusters
 
-`mongodbatlas_cluster` describes all Advanced Clusters by the provided project_id. The data source requires your Project ID.
+`mongodbatlas_advanced_cluster` describes all Advanced Clusters by the provided project_id. The data source requires your Project ID.
```
Comment (lantoli) [inline docs/data-sources/advanced_clusters.md]: header for adv_cluster plural ds was wrong

File: scripts/tflint.sh
Comment (lantoli) [inline scripts/tflint.sh:1]: no need to iterate through all the examples using recursive option, done in makefile goal directly

File: docs/data-sources/data_lake_pipeline.md [hunk 6]
```diff
@@ -38,7 +38,7 @@ resource "mongodbatlas_data_lake_pipeline" "pipeline" {
 
       source {
         type = "ON_DEMAND_CPS"
-        cluster_name = mongodbatlas_cluster.automated_backup_test.name
+        cluster_name = mongodbatlas_advanced_cluster.automated_backup_test.name
```
Comment (EspenAlbert) [inline docs/data-sources/data_lake_pipeline.md:48]: missing migration above

File: docs/resources/advanced_cluster.md [hunk 7]
```diff
@@ -303,29 +303,29 @@ resource "mongodbatlas_advanced_cluster" "cluster" {
 Standard
 ```terraform
 output "standard" {
-    value = mongodbatlas_cluster.cluster-test.connection_strings[0].standard
+    value = mongodbatlas_advanced_cluster.cluster.connection_strings[0].standard
```
Comment (EspenAlbert) [inline docs/resources/advanced_cluster.md:306]: nice fixes!

File: docs/resources/data_lake_pipeline.md [hunk 8]
```diff
@@ -38,7 +38,7 @@ resource "mongodbatlas_data_lake_pipeline" "pipeline" {
 
       source {
         type = "ON_DEMAND_CPS"
-        cluster_name = mongodbatlas_cluster.automated_backup_test.name
+        cluster_name = mongodbatlas_advanced_cluster.automated_backup_test.name
```
Comment (EspenAlbert) [inline docs/resources/data_lake_pipeline.md:48]: missing migration above

File: examples/mongodbatlas_encryption_at_rest/aws/atlas-cluster/README.md [hunk 9]
```diff
@@ -70,15 +70,15 @@ terraform destroy
 
 1. Import the cluster using the Project ID and cluster name (e.g. `5beae24579358e0ae95492af-MyCluster`):
 
-        $ terraform import mongodbatlas_cluster.my_cluster ProjectId-ClusterName
+        $ terraform import mongodbatlas_advanced_cluster.my_cluster ProjectId-ClusterName
```
Comment (EspenAlbert) [inline examples/mongodbatlas_encryption_at_rest/aws/atlas-cluster/README.md]: Error from before (also on line 75, 78, and 82):
`my_cluster` -> `cluster`

File: examples/mongodbatlas_network_peering/azure/variables.tf [hunk 10]
```diff
@@ -7,12 +7,6 @@ variable "private_key" {
 variable "project_id" {
   type = string
 }
-variable "provider_instance_size_name" {
```
Comment (EspenAlbert) [inline examples/mongodbatlas_network_peering/azure/variables.tf:10]: why was this removed?

File: examples/mongodbatlas_privatelink_endpoint/azure/main.tf [hunk 11]
```diff
@@ -19,12 +19,12 @@ resource "azurerm_virtual_network" "test" {
...
-  resource_group_name                            = var.resource_group_name
-  virtual_network_name                           = azurerm_virtual_network.test.name
-  address_prefixes                               = ["10.0.1.0/24"]
-  enforce_private_link_service_network_policies  = true
-  enforce_private_link_endpoint_network_policies = true
+  name                                          = "testsubnet"
+  resource_group_name                           = var.resource_group_name
+  virtual_network_name                          = azurerm_virtual_network.test.name
+  address_prefixes                              = ["10.0.1.0/24"]
+  private_link_service_network_policies_enabled = true
```
Comment (EspenAlbert) [inline examples/mongodbatlas_privatelink_endpoint/azure/main.tf:26]: bonus fixes?

File: examples/starter/variables.tf [hunk 12]
```diff
@@ -26,10 +26,6 @@ variable "region" {
   type        = string
   description = "MongoDB Atlas Cluster Region, must be a region for the provider given"
 }
-variable "mongodbversion" {
```
Comment (EspenAlbert) [inline examples/starter/variables.tf:29]: why remove this?

File: .github/workflows/examples.yml [hunk 13]
```diff
@@ -15,52 +15,27 @@ env:
...
-        run: go build -o terraform-plugin-dir/terraform-provider-mongodbatlas-dev_v99.99.99_x5 .
-      - name: override plugin
```
Comment (AgustinBettati) [inline .github/workflows/examples.yml:33]: Can you clarify how the override of our provider is handled now?

File: examples/mongodbatlas_api_key/create-and-assign-pak-together/versions.tf (hunk 3 above)
Comment (AgustinBettati) [reply examples/mongodbatlas_api_key/create-and-assign-pak-together/versions.tf:5]: is it possible that for versions below 1.13.2 this examples does not work? Cant recall in what version we made changes to the pak resources.

**Review by AgustinBettati — APPROVED** [review]
LGTM, small doubt on version constraint that was explicit to be above 1.13.2

File: .github/workflows/examples.yml (hunk 13 above)
Comment (lantoli) [reply .github/workflows/examples.yml:33]: it uses the recommended way: https://developer.hashicorp.com/terraform/cli/config/config-file#development-overrides-for-provider-developers
that we also recommend from our guides: https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/development-setup.md#building

(i've also added the info in the PR description)

**Comment by lantoli** [comment]
> is it possible that for versions below 1.13.2 this examples does not work? Cant recall in what version we made changes to the pak resources.

to be honest I don't remember, the problem with version = "~> 1.13.2" is that for instance 1.14 is not valid, whereas with ~> 1.0 the latest version can be used

i've seen "~> 1.13.2"  in a few examples so I don't know if it was because it was the current version when the example was created

edit:  right now the example won't ever fail using "~> 1.0" as it'll get the latest provider version (e.g. 1.17.3)

File: docs/data-sources/data_lake_pipeline.md (hunk 6 above)
Comment (lantoli) [reply docs/data-sources/data_lake_pipeline.md:48]: thx, updated

File: docs/resources/data_lake_pipeline.md (hunk 8 above)
Comment (lantoli) [reply docs/resources/data_lake_pipeline.md:48]: thx, changed

File: examples/mongodbatlas_encryption_at_rest/aws/atlas-cluster/README.md (hunk 9 above)
Comment (lantoli) [reply examples/mongodbatlas_encryption_at_rest/aws/atlas-cluster/README.md]: thx! changed

File: examples/mongodbatlas_network_peering/azure/variables.tf (hunk 10 above)
Comment (lantoli) [reply examples/mongodbatlas_network_peering/azure/variables.tf:10]: good catch, reverted

File: examples/mongodbatlas_privatelink_endpoint/azure/main.tf (hunk 11 above)
Comment (lantoli) [reply examples/mongodbatlas_privatelink_endpoint/azure/main.tf:26]: yeah, replacing some deprecated attributes

File: examples/starter/variables.tf (hunk 12 above)
Comment (lantoli) [reply examples/starter/variables.tf:29]: because it was not important for the example, so I tried to keep the cluster creation only with relevant attributes for the examples

**Review by EspenAlbert — APPROVED** [review]
Fantastic!

**Comment by lantoli** [comment]
@EspenAlbert thanks for the offline discussion!
